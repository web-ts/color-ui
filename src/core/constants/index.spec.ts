import { it, expect, describe } from "vitest";
import { DEFAULT_COLORS, DEFAULT_COLOR_UI_OPTIONS, DEFAULT_ICONS } from ".";

describe("constants", () => {
  it("constants are matching the defaults", () => {
    expect(DEFAULT_COLORS).toMatchSnapshot();
    expect(DEFAULT_ICONS).toMatchSnapshot();
    expect(DEFAULT_COLOR_UI_OPTIONS).toMatchSnapshot();
  });
});
