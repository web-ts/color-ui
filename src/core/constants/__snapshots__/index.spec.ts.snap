// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`constants > constants are matching the defaults 1`] = `
{
  "dark": {
    "background": "#2c3e50",
    "danger": "#e74c3c",
    "info": "#3498db",
    "primary": "#1abc9c",
    "secondary": "#bdc3c7",
    "success": "#2ecc71",
    "tertiary": "#9b59b6",
    "text": {
      "dark": "#090909",
      "light": "#fefefe",
    },
    "warning": "#f1c40f",
  },
  "light": {
    "background": "#ecf0f1",
    "danger": "#e74c3c",
    "info": "#2980b9",
    "primary": "#3498db",
    "secondary": "#7f8c8d",
    "success": "#27ae60",
    "tertiary": "#2ecc71",
    "text": {
      "dark": "#090909",
      "light": "#fefefe",
    },
    "warning": "#f39c12",
  },
}
`;

exports[`constants > constants are matching the defaults 2`] = `
{
  "close": "ic:round-close",
  "danger": "ic:round-dangerous",
  "info": "ic:round-info",
  "menu": "ic:round-menu",
  "spinner": "line-md:loading-twotone-loop",
  "success": "ic:round-check-circle",
  "warning": "ic:round-warning",
}
`;

exports[`constants > constants are matching the defaults 3`] = `
{
  "colors": {
    "dark": {
      "background": "#2c3e50",
      "danger": "#e74c3c",
      "info": "#3498db",
      "primary": "#1abc9c",
      "secondary": "#bdc3c7",
      "success": "#2ecc71",
      "tertiary": "#9b59b6",
      "text": {
        "dark": "#090909",
        "light": "#fefefe",
      },
      "warning": "#f1c40f",
    },
    "light": {
      "background": "#ecf0f1",
      "danger": "#e74c3c",
      "info": "#2980b9",
      "primary": "#3498db",
      "secondary": "#7f8c8d",
      "success": "#27ae60",
      "tertiary": "#2ecc71",
      "text": {
        "dark": "#090909",
        "light": "#fefefe",
      },
      "warning": "#f39c12",
    },
  },
  "iconProvider": {
    "component": null,
    "defaultProps": {},
    "keyPropName": "icon",
  },
  "icons": {
    "close": "ic:round-close",
    "danger": "ic:round-dangerous",
    "info": "ic:round-info",
    "menu": "ic:round-menu",
    "spinner": "line-md:loading-twotone-loop",
    "success": "ic:round-check-circle",
    "warning": "ic:round-warning",
  },
}
`;
